app/scripts/modules/core/src/pipeline/filter/executionFilter.service.ts

```diff
@@ -82,6 +82,40 @@ export class ExecutionFilterService {
     this.groupsUpdatedStream.next(groups);
   }
 
+  public static doesPipelineMatchCheckedCategories(config: IPipeline, checkedCategories: string[]): boolean {
+    if (checkedCategories.length === 0 || !config.categories || config.categories.length === 0) {
+      return false;
+    }
+    const decoded: IPipelineCategory[] = checkedCategories
+      .map((encoded) => encoded.split(':').map(decodeURIComponent))
+      .map(([name, value]) => ({ name, value }));
+    const grouped = groupBy(decoded, 'name');
+    const groups = Object.keys(grouped);
+    // We use .every() to logically AND the different categories
+    return groups.every((group) => {
+      const checkedValues = grouped[group].map((category) => category.value);
+      const relevantValues = (config.categories || [])
+        .filter((category) => category.name === group)
+        .map((category) => category.value);
+      return checkedValues.some((checkedValue) => relevantValues.includes(checkedValue));
```

Comment:
`.some()`

am I reading this right that we AND the categories, but OR the selected values?

Reply:
Yes.

We OR the values because they are assumed to be mutually exclusive and AND the categories (names) because they're assumed to be different dimensions.

This is analogous to clusters where, for instance, we AND between account and region, but we OR the values of those dimensions.

Reply:
makes sense!